Modules/System Tests/Location/src/LocationTest.Codeunit.al

```diff
@@ -0,0 +1,157 @@
+// ------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// ------------------------------------------------------------------------------------------------
+
+codeunit 50113 "Location Test"
+{
+    // [FEATURE] [Location] 
+
+    Subtype = Test;
+    TestPermissions = Disabled;
+
+    var
+        Assert: Codeunit "Library Assert";
+
+    [Test]
+    [Scope('OnPrem')]
+    procedure TestIsLocationAvailable()
+    var
+        Location: Page Location;
+        LocationTestLibrary: Codeunit "Location Test Library";
+    begin
+        // [When] Location test library subscribers are binded.
+        BindSubscription(LocationTestLibrary);
+
+        // [Then] The return value of IsAvailable is 'true'.
+        Assert.IsTrue(Location.IsAvailable(), 'The location should be available during testing.');
+    end;
+
+    [Test]
+    [Scope('OnPrem')]
+    [HandlerFunctions('LocationPageHandler')]
+    procedure TestHasLocationSuccess()
+    var
+        Location: Page Location;
+        LocationTestLibrary: Codeunit "Location Test Library";
+        ActualLatitude: Decimal;
+        ActualLongitude: Decimal;
+        ExpectedLatitude: Decimal;
+        ExpectedLongitude: Decimal;
+    begin
+        // [Given] Location test library subscribers are binded.
+        BindSubscription(LocationTestLibrary);
+
+        // [When] The location page is run as modal.
+        Location.RunModal();
+
+        // [Then] The location page has a location 
+        Assert.IsTrue(Location.HasLocation(), 'The location page has a location');
+    end;
+
+    [Test]
+    [Scope('OnPrem')]
+    [HandlerFunctions('LocationPageHandler')]
+    procedure TestGetLocationSuccesss()
+    var
+        Location: Page Location;
```

Comment:
Use TestPage instead of Page

Reply:
> Use TestPage instead of Page

I am primarily testing the functions that are on the page. I cannot find any example of how I would do that if I am using a TestPage instead. The camera module is also using a Page to test the functions.